packages/wouter/src/index.js

```diff
@@ -204,17 +204,29 @@ export const Link = forwardRef((props, ref) => {
     }
   });
 
-  // wraps children in `a` if needed
-  const extraProps = {
-    // handle nested routers and absolute paths
-    href: href[0] === "~" ? href.slice(1) : router.base + href,
+  // handle nested routers and absolute paths
+  const link = href[0] === "~" ? href.slice(1) : router.base + href;
+
+  if (asChild) {
+    if (!isValidElement(children)) {
+      throw Error("Only one child allowed");
```

Comment:
I'm not sure about `error` here. is it okay to trow error when children are invalid?

@molefrog what do you think?